lib/run_part.c

```diff
@@ -61,7 +61,7 @@ int run_parts (const char *directory, const char *name, const char *action)
 		struct stat  sb;
 
 		if (asprintf(&s, "%s/%s", directory, namelist[n]->d_name) == -1) {
-			fprintf (shadow_logfd, "could not allocate memory\n");
+			fprintf (shadow_logfd, "asprintf: %s\n", strerror(errno));
```

Comment:
Same here, if not I fear it'll be difficult to understand the problem. Example `asprintf failed...`

Reply:
```
asprintf: Cannot allocate memory
```

Should imply a failure.  :)

Reply:
You are right here